src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -21,6 +21,18 @@
 #include <unistd.h>
 
 #include "xcl_perfmon_parameters.h"
+#define SEND_RESP2QDMA()\
+    if (buf_size == 0) {\
+    	buf = malloc(ri_msg->size());\
```

Comment:
memory leak due to buf pointer

Reply:
This buf gets de-allocated in destructor of Q2H_helper class

Reply:
This looks very unsafe.   Why is a raw pointer used?   Should it be std::vector<char*> to manage buf, then use buf.data() to get the raw managed pointer?

Reply:
Smart pointer is used to create buffer